nutrition_labels/create_training_data.py

```diff
@@ -0,0 +1,205 @@
+- Clearly label which data came from which source (EPMC, RF, Grants (not using Prodigy), Prodigy grants)
+- Filter out any sources not wanted
+- Output this as the training data
+
+"""
+
+from argparse import ArgumentParser
+import configparser
+import os
+from datetime import datetime
+import json
+import ast
+
+import pandas as pd
+import numpy as np
+
+from nutrition_labels.grant_data_processing import (
+    load_process_data_sources,
+    merge_grants_sources,
+    clean_grants_data,
+    deduplicate_similar_grants,
+)
+from nutrition_labels.prodigy_training_data import load_prodigy_tags
+
+SOURCES = ["RF", "EPMC", "Grants", "Prodigy grants"]
+
+
+def output_counts(training_data):
+
+    print("Number of training data points from the various sources:")
+    multi_tag_data = training_data[pd.notnull(training_data[SOURCES]).sum(axis=1) != 1]
+    print(f"Multiple sources: {len(multi_tag_data)}")
+    single_tag_data = training_data[pd.notnull(training_data[SOURCES]).sum(axis=1) == 1]
+    num_source = pd.notnull(single_tag_data[SOURCES]).sum(axis=0)
+    print(f"Individual sources: \n{num_source}")
+
+
+def combine_original_prodigy(training_data_original, prodigy_output):
+    """
+    Combine the original training data using EPMC, RF and grants
+    with the data tagged in Prodigy, in a format which makes it clear
+    which data point came from each source
+    """
+
+    training_data_original_dict = {}
+    for i, row in training_data_original.iterrows():
+        training_data_original_dict[row["Internal ID"]] = {
+            "Original tag": row["Relevance code"],
+            "RF": None if pd.isnull(row["Normalised code - RF"]) else 1,
+            "EPMC": None if pd.isnull(row["Normalised code - EPMC"]) else 1,
+            "Grants": None
+            if pd.isnull(row["Normalised code - grants"])
+            else (0 if int(row["Normalised code - grants"]) == 5 else 1),
+            "Prodigy grants": None,
+        }
+
+    # Add Prodigy tags to original tags
+    for grant_number, tech_cat in prodigy_output.items():
+        # Add this grant number if not in the original data
+        if not training_data_original_dict.get(grant_number):
+            training_data_original_dict[grant_number] = {
+                "Original tag": None,
+                "RF": None,
+                "EPMC": None,
+                "Grants": None,
+                "Prodigy grants": tech_cat,
+            }
+
+    training_data = pd.DataFrame(training_data_original_dict).T
+    training_data["Internal ID"] = training_data.index
+
+    return training_data
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--config_path",
+        help="Path to config file",
+        default="configs/training_data/2021.03.08.ini",
+    )
+
+    args = parser.parse_args()
+
+    config = configparser.ConfigParser()
+    config.read(args.config_path)
+
+    # Load Prodigy data
+    use_prodigy_directly = False
```

Comment:
Note that you have a lot of functionality outside of functions and inside the `__name__ == __main__` section. 

These are untested so the only way you know they are working and not broken is to run them. I am assuming that is what you do to ensure this script works. It would be better to move them to functions and add tests.